icu4c/source/test/intltest/tztest.cpp

```diff
@@ -1171,36 +1171,56 @@ void TimeZoneTest::TestCustomParse()
-        {"GMT+15:99", kUnparseable},
-        {"GMT+",      kUnparseable},
-        {"GMT-",      kUnparseable},
-        {"GMT+0:",    kUnparseable},
-        {"GMT-:",     kUnparseable},
-        {"GMT-YOUR.AD.HERE",    kUnparseable},
-        {"GMT+0010",  (10*60)}, // Interpret this as 00:10
-        {"GMT-10",    (-10*60*60)},
-        {"GMT+30",    kUnparseable},
-        {"GMT-3:30",  (-(3*60+30)*60)},
-        {"GMT-230",   (-(2*60+30)*60)},
-        {"GMT+05:13:05",    ((5*60+13)*60+5)},
-        {"GMT-71023",       (-((7*60+10)*60+23))},
-        {"GMT+01:23:45:67", kUnparseable},
-        {"GMT+01:234",      kUnparseable},
-        {"GMT-2:31:123",    kUnparseable},
-        {"GMT+3:75",        kUnparseable},
-        {"GMT-01010101",    kUnparseable},
+        {u"GMT",       kUnparseable},   //Isn't custom. [returns normal GMT]
+        {u"GMT-YOUR.AD.HERE", kUnparseable},
+        {u"GMT0",      kUnparseable},
+        {u"GMT+0",     (0)},
+        {u"GMT+1",     (1*60*60)},
+        {u"GMT-0030",  (-30*60)},
+        {u"GMT+15:99", kUnparseable},
+        {u"GMT+",      kUnparseable},
+        {u"GMT-",      kUnparseable},
+        {u"GMT+0:",    kUnparseable},
+        {u"GMT-:",     kUnparseable},
+        {u"GMT-YOUR.AD.HERE",    kUnparseable},
+        {u"GMT+0010",  (10*60)}, // Interpret this as 00:10
+        {u"GMT-10",    (-10*60*60)},
+        {u"GMT+30",    kUnparseable},
+        {u"GMT-3:30",  (-(3*60+30)*60)},
+        {u"GMT-230",   (-(2*60+30)*60)},
+        {u"GMT+05:13:05",    ((5*60+13)*60+5)},
+        {u"GMT-71023",       (-((7*60+10)*60+23))},
+        {u"GMT+01:23:45:67", kUnparseable},
+        {u"GMT+01:234",      kUnparseable},
+        {u"GMT-2:31:123",    kUnparseable},
+        {u"GMT+3:75",        kUnparseable},
+        {u"GMT-01010101",    kUnparseable},
+        {u"GMT-4E58",        kUnparseable}, // ICU-22637
+        {u"GMT-4e58",        kUnparseable}, // ICU-22637
+        {u"GMT-1E01",        kUnparseable}, // ICU-22637
+        {u"GMT-2E01",        kUnparseable}, // ICU-22637
+        {u"GMT-2e01",        kUnparseable}, // ICU-22637
+        {u"GMT-9e02",        kUnparseable}, // ICU-22637
+        {u"GMT-1e03",        kUnparseable}, // ICU-22637
+        {u"GMT-2e03",        kUnparseable}, // ICU-22637
+        {u"GMT-500M",        kUnparseable}, // ICU-22637
+        {u"GMT-500T",        kUnparseable}, // ICU-22637
+        {u"GMT-9E00",        kUnparseable}, // ICU-22637
+        {u"GMT-0X0F",        kUnparseable}, // ICU-22637
+        {u"GMT-0x0F",        kUnparseable}, // ICU-22637
+        {u"GMT-0x12",        kUnparseable}, // ICU-22637
+        {u"GMT-B111",        kUnparseable}, // ICU-22637
+        {u"GMT-b111",        kUnparseable}, // ICU-22637
+        {u"GMT-0b11",        kUnparseable}, // ICU-22637
+        {u"GMT-๑๒",          kUnparseable}, // ICU-22637
+        {u"GMT-๑๒:๓๔",       kUnparseable}, // ICU-22637
+        {u"GMT+๑๒:๓๔:๕๖",    kUnparseable}, // ICU-22637
```

Comment:
Why are we rejecting non-ASCII digits? We had been accepting them and they could reasonably appear in a localized string.

Reply:
yes, you are correct. This is my bad. I have a misunderstanding of the prior behavior. I should change that back.

Reply:
Please review my fix in https://github.com/unicode-org/icu/pull/2803